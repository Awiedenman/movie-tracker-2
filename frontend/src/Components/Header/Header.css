.active {
  border: 1px solid palevioletred;
}