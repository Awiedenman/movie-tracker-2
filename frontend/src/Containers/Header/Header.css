.header {
  background: #000;
  padding: 2rem;
}

.header__home-link {
  color: var(--primary-red);
  font-size: 2.5rem;
  text-decoration: none;
}

.header__home-link h1 {
  font-weight: 100;
}

.header__nav  a,
.sign-out {
  color: var(--primary-white);
  cursor: pointer;
  font-size: 1.5rem;
  margin-right: 2rem;
  padding: 1rem;
  text-decoration: none;
}

.header__welcome-message {
  color: var(--primary-white);
  font-size: 1.5rem;
  padding-bottom: 2rem;
}

.active {
  background: var(--primary-red);
  border: 1px solid var(--primary-red);
  color: var(--primary-white);
  transition: all .2s ease-in-out;
}

.sign-out,
.favorites {
  border: 1px solid var(--primary-red);
  color: var(--primary-white);
  transition: all .2s ease-in-out;
}

.sign-out:hover,
.favorites:hover {
  background: var(--primary-red);
}