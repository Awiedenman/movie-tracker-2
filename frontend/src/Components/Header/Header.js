import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import './Header.css';

const Header = () => {
  return (
    <header className="header">
      <h1>Movie Tracker</h1>
      <nav>
        <NavLink exact to="/login">Sign In</NavLink>
        <NavLink to="/sign-up">Sign Up</NavLink>
      </nav>
    </header>
  );
};

Header.propTypes = {

};

export default Header;
