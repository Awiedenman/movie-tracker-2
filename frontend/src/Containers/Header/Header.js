import React from 'react';
import { NavLink, Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { userSignOut } from '../../Actions/index';
import PropTypes from 'prop-types';

import './Header.css';

export const Header = props => {
  const { user, signOut } = props;

  return (
    <header className="header">
      <Link exact to="/">
        <h1>Movie Tracker</h1>
      </Link>
      <nav>
        {user.name && <p>Welcome back {user.name}</p>}
        {
          user.name ?
            <button className="sign-out" onClick={() => signOut()}>Sign Out</button>
            :
            <div>
              <NavLink exact to="/login">Sign In</NavLink>
              <NavLink exact to="/sign-up">Sign Up</NavLink>
            </div>

        }
      </nav>
    </header>
  );
};

Header.propTypes = {
  user: PropTypes.object,
  signOut: PropTypes.func
};

export const mapStateToProps = state => ({
  user: state.userInfo
});

export const mapDispatchToProps = dispatch => ({
  signOut: () => dispatch(userSignOut())
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header));
