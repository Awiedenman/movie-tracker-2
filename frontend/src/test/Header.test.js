import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Components/Header/Header';

describe('Header', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<Header />));
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
