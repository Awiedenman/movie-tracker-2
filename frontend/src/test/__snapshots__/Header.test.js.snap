// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    signOut={[Function]}
    user={
        Object {
            "id": 1,
            "name": "Bob",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
>
          <h1>
                    Movie Tracker
          </h1>
</Link>,
        <Link
          replace={false}
          to="favorites"
>
          favorites
</Link>,
        <nav>
          <p>
                    Welcome back 
                    Bob
          </p>
          <button
                    className="sign-out"
                    onClick={[Function]}
          >
                    Sign Out
          </button>
</nav>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1>
            Movie Tracker
</h1>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Movie Tracker",
          },
          "ref": null,
          "rendered": "Movie Tracker",
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "favorites",
          "replace": false,
          "to": "favorites",
        },
        "ref": null,
        "rendered": "favorites",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Welcome back 
              Bob
</p>,
            false,
            <button
              className="sign-out"
              onClick={[Function]}
>
              Sign Out
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Welcome back ",
                "Bob",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Welcome back ",
              "Bob",
            ],
            "type": "p",
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Out",
              "className": "sign-out",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign Out",
            "type": "button",
          },
        ],
        "type": "nav",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
>
            <h1>
                        Movie Tracker
            </h1>
</Link>,
          <Link
            replace={false}
            to="favorites"
>
            favorites
</Link>,
          <nav>
            <p>
                        Welcome back 
                        Bob
            </p>
            <button
                        className="sign-out"
                        onClick={[Function]}
            >
                        Sign Out
            </button>
</nav>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h1>
              Movie Tracker
</h1>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Movie Tracker",
            },
            "ref": null,
            "rendered": "Movie Tracker",
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "favorites",
            "replace": false,
            "to": "favorites",
          },
          "ref": null,
          "rendered": "favorites",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Welcome back 
                Bob
</p>,
              false,
              <button
                className="sign-out"
                onClick={[Function]}
>
                Sign Out
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Welcome back ",
                  "Bob",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Welcome back ",
                "Bob",
              ],
              "type": "p",
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Out",
                "className": "sign-out",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Out",
              "type": "button",
            },
          ],
          "type": "nav",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Header should match when logged out 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
>
          <h1>
                    Movie Tracker
          </h1>
</Link>,
        <Link
          replace={false}
          to="favorites"
>
          favorites
</Link>,
        <nav>
          <div>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              exact={true}
                              to="/login"
                    >
                              Sign In
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              exact={true}
                              to="/sign-up"
                    >
                              Sign Up
                    </NavLink>
          </div>
</nav>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1>
            Movie Tracker
</h1>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Movie Tracker",
          },
          "ref": null,
          "rendered": "Movie Tracker",
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "favorites",
          "replace": false,
          "to": "favorites",
        },
        "ref": null,
        "rendered": "favorites",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            undefined,
            <div>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            exact={true}
                            to="/login"
              >
                            Sign In
              </NavLink>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            exact={true}
                            to="/sign-up"
              >
                            Sign Up
              </NavLink>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  exact={true}
                  to="/login"
>
                  Sign In
</NavLink>,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  exact={true}
                  to="/sign-up"
>
                  Sign Up
</NavLink>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Sign In",
                  "exact": true,
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Sign In",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Sign Up",
                  "exact": true,
                  "to": "/sign-up",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "nav",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
>
            <h1>
                        Movie Tracker
            </h1>
</Link>,
          <Link
            replace={false}
            to="favorites"
>
            favorites
</Link>,
          <nav>
            <div>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    exact={true}
                                    to="/login"
                        >
                                    Sign In
                        </NavLink>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    exact={true}
                                    to="/sign-up"
                        >
                                    Sign Up
                        </NavLink>
            </div>
</nav>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h1>
              Movie Tracker
</h1>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Movie Tracker",
            },
            "ref": null,
            "rendered": "Movie Tracker",
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "favorites",
            "replace": false,
            "to": "favorites",
          },
          "ref": null,
          "rendered": "favorites",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
              <div>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                exact={true}
                                to="/login"
                >
                                Sign In
                </NavLink>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                exact={true}
                                to="/sign-up"
                >
                                Sign Up
                </NavLink>
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    exact={true}
                    to="/login"
>
                    Sign In
</NavLink>,
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    exact={true}
                    to="/sign-up"
>
                    Sign Up
</NavLink>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Sign In",
                    "exact": true,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Sign In",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Sign Up",
                    "exact": true,
                    "to": "/sign-up",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "nav",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
